api/public.proto

```diff
@@ -794,10 +877,88 @@ message Trip {
   }
 }
 
+// The Vehicle resource.
+//
+// This resource corresponds to the [vehicle position type in the GTFS static
+// specification](https://developers.google.com/transit/gtfs-realtime/reference#message-vehicleposition).
 message Vehicle {
+  // A unique ID for the vehicle.
+  string id = 1;
+
+  // A reference to the vehicle's trip.
+  optional Trip.Reference trip = 2;
+
+  // The vehicle's current latitude.
+  optional double latitude = 3;
+
+  // The vehicle's current longitude.
+  optional double longitude = 4;
+
+  // The vehicle's current bearing.
+  optional float bearing = 5;
+
+  // The vehicle's current odometer reading.
+  optional double odometer = 6;
+
+  // The vehicle's current speed.
+  optional float speed = 7;
+
+  // The stop sequence index of the vehicle's current stop.
+  optional int32 stop_sequence = 8;
+
+  // A reference to the vehicle's current stop.
+  optional Stop.Reference stop = 9;
+
+  // Corresponds to [VehicleStopStatus](https://developers.google.com/
+  // transit/gtfs-realtime/reference#enum-vehiclestopstatus).
+  enum CurrentStatus {
+    INCOMING_AT = 0;
+    STOPPED_AT = 1;
+    IN_TRANSIT_TO = 2;
+  }
+  // The vehicle's current status.
+  optional CurrentStatus current_status = 10;
+
+  // The timestamp of the last update to the vehicle's position.
+  optional int64 updated_at = 11;
+
+  // Corresponds to [CongestionLevel](https://developers.google.com/
+  // transit/gtfs-realtime/reference#enum-congestionlevel).
+  enum CongestionLevel {
+    UNKNOWN_CONGESTION_LEVEL = 0;
+    RUNNING_SMOOTHLY = 1;
+    STOP_AND_GO = 2;
+    CONGESTION = 3;
+    SEVERE_CONGESTION = 4;
+  }
+  // The vehicle's current congestion level.
+  CongestionLevel congestion_level = 12;
+
+  // Corresponds to [OccupancyStatus](https://developers.google.com/
+  // transit/gtfs-realtime/reference#enum-occupancystatus).
+  enum OccupancyStatus {
+    EMPTY = 0;
+    MANY_SEATS_AVAILABLE = 1;
+    FEW_SEATS_AVAILABLE = 2;
+    STANDING_ROOM_ONLY = 3;
+    CRUSHED_STANDING_ROOM_ONLY = 4;
+    FULL = 5;
+    NOT_ACCEPTING_PASSENGERS = 6;
+  }
+  // The vehicle's current occupancy status.
+  optional OccupancyStatus occupancy_status = 13;
+
+  // The percentage of seats occupied.
+  optional int32 occupancy_percentage = 14;
+
   // Reference is the reference type for the vehicle resource.
   message Reference {
     string id = 1;
+    Resource resource = 2;
+    optional double latitude = 3;
+    optional double longitude = 4;
+    optional float bearing = 5;
+    optional int64 updated_at = 6;
```

Comment:
My thought here was these would be the properties that a consumer of trips would be most likely interested in. For example, if you get a trip, you can quickly also pinpoint the vehicle on a map without another API call. This is sort of similar to trip references, which also has some commonly needed properties.

I am totally fine either removing these fields or only including them with a query param option if you think this is too much data for a reference.

Reply:
Yeah I would prefer not to have them if that was okay? We could always add them back in down the line if it turned out to make consuming the API much easier? 

Another thing we could do is add a vehicle reference as a field to the trip reference - thus in the GetStops view, say, we could go directly to GetVehicle rather than having to poll GetTrip first. (I understand I'm somewhat contradicting myself by proposing putting another field on the trip reference while arguing against fields on the vehicle reference...)

Reply:
Sounds good - I simplified vehicle references to just be the id/resource of the vehicle in the latest iteration.